[Osquery] Add docs for Osquery API (#137162)
elastic/kibana: docs/api/osquery-manager/live-queries/get-all.asciidoc

[[osquery-manager-live-queries-api-get-all]]
=== Get live queries API
++++
<titleabbrev>Get live queries</titleabbrev>
++++

experimental[] Get live queries.

[[osquery-manager-live-queries-api-get-all-request]]
==== Request

`GET <kibana host>:<port>/api/osquery/live_queries`

`GET <kibana host>:<port>/s/<space_id>/api/osquery/live_queries`

[[osquery-manager-live-queries-api-get-all-params]]
==== Path parameters

`space_id`::
(Optional, string) An identifier for the space. If `space_id` is not provided in the URL, the default space is used.

=== Query parameters

`page`::
(Optional, integer) The page number to return. Defaults to `1`.

`perPage`::
(Optional, integer) The number of rules to return per page. Defaults to `20`.

`sortField`::

> (Optional, string) Determines which field is used to sort the results,

Reviewer: ```suggestion
(Optional, string) The field that is used to sort the results. Options include `createdAt` or `updatedAt`.
```

> `createdAt` or `updatedAt`. Defaults to `createdAt`.

Reviewer: ```suggestion
The default is `createdAt`.
```

+
NOTE: Even though the JSON case object uses `created_at` and `updated_at`
fields, you must use `createdAt` and `updatedAt` fields in the URL
query.

`sortOrder`::

> (Optional, string) Determines the sort order, which can be `desc` or `asc`.

Reviewer: ```suggestion
(Optional, string) Specified the sort order. Options include `desc` or `asc`.
```

> Defaults to `desc`.

Reviewer: ```suggestion
The defaults is `desc`.
```

[[osquery-manager-live-queries-api-get-all-codes]]
==== Response code

`200`::
Indicates a successful call.

[[osquery-manager-live-queries-api-get-all-example]]
==== Example

> Retrieve last 10 live queries :

Reviewer: ```suggestion
Retrieve the last 10 live queries :
```

[source,sh]
--------------------------------------------------
$ curl -X GET api/osquery/live_queries?page=1&perPage=10
--------------------------------------------------
// KIBANA

> The API returns a JSON object listing the retrieved live queries. For example:

Reviewer: ```suggestion
The API returns a JSON object of the retrieved live queries:
```
